torch/_inductor/runtime/benchmarking.py

```diff
@@ -49,50 +48,76 @@ def benchmark(
         fn_kwargs: Dict[str, Any],
         **kwargs: Any,
     ) -> float:
-        """Construct benchmarkable callable and dispatch benchmark request to the appropriate
-        benchmarking function depending on the device type of `fn_args` and `fn_kwargs`.
+        """Benchmark `fn(*fn_args, *fn_kwargs)` and return the runtime, in milliseconds (the
+        actual runtime calculation is dictated by the benchmarking implementation, but may be
+        one of [mean, median, minimum, etc.]). Functions as a convenience wrapper around
+        device-specific implementations, like `benchmark_cpu` and `benchmark_gpu`. Raises
+        `ValueError(...)` if we can't safely infer the device type of `fn`; for example,
+        if multiple device types are found in `fn_args` and `fn_kwargs`, or if no device
+        types are found.
 
         Arguments:
         - fn: The function to benchmark.
         - fn_args: The function's arguments.
         - fn_kwargs: The function's kwargs.
 
         Keyword Arguments:
-        - **kwargs: The benchmarker's keyword arguments.
+        - **kwargs: The benchmarking implementation's kwargs.
 
         Returns:
         - The runtime of `fn(*fn_args, **fn_kwargs)`, in milliseconds.
         """
-        if is_cpu_device(list(fn_args) + list(fn_kwargs.values())):
-            return self.benchmark_cpu(lambda: fn(*fn_args, **fn_kwargs), **kwargs)
-        return self.benchmark_gpu(lambda: fn(*fn_args, **fn_kwargs), **kwargs)
+        inferred_device = None
+        for arg_or_kwarg in fn_args + tuple(fn_kwargs.items()):
```

Comment:
`tuple(fn_kwargs.items()` will iterate over a tuples of (key, value) so we'll never see Tensros here. also, better not to materialize a tuple, you can do `itertools.chain(fn_args, fn_kwargs.keys(), fn_kwargs.values())`

Reply:
yep, should be `fn_kwargs.values()`. Will use chain